ui/src/app/machines/views/MachineList/MachineListHeader/ActionFormWrapper/DeployForm/DeployFormFields/UserDataField/UserDataField.test.tsx

```diff
@@ -0,0 +1,244 @@
+import React from "react";
+import { Provider } from "react-redux";
+
+import DeployForm from "../../DeployForm";
+import { TSFixMe } from "app/base/types";
+
+const mockStore = configureStore();
+
+class MockFileReader {
+  result: string;
+  constructor() {
+    this.result = "test file content";
+  }
+  /* eslint-disable @typescript-eslint/no-empty-function */
+  onabort() {}
+  onerror() {}
+  onload() {}
+  /* eslint-enable @typescript-eslint/no-empty-function */
+  readAsText() {
+    this.onload();
+  }
+}
+
+const createFile = (
+  name: string,
+  size: number,
+  type: string,
+  contents = ""
+) => {
+  const file = new File([contents], name, { type });
+  Reflect.defineProperty(file, "size", {
+    get() {
+      return size;
+    },
+  });
+  return file;
+};
+
+describe("DeployFormFields", () => {
+  let state: TSFixMe;
+  let wrapper: ReactWrapper;
+
+  beforeEach(async () => {
+    state = {
+      config: {
+        items: [
+          {
+            name: "default_osystem",
+            value: "ubuntu",
+            choices: [
+              ["centos", "CentOS"],
+              ["ubuntu", "Ubuntu"],
+            ],
+          },
+        ],
+        errors: {},
+        loaded: true,
+        loading: false,
+      },
+      general: {
+        defaultMinHweKernel: {
+          data: "",
+          errors: {},
+          loaded: true,
+          loading: false,
+        },
+        osInfo: {
+          data: {
+            osystems: [
+              ["centos", "CentOS"],
+              ["ubuntu", "Ubuntu"],
+            ],
+            releases: [
+              ["centos/centos66", "CentOS 6"],
+              ["centos/centos70", "CentOS 7"],
+              ["ubuntu/bionic", 'Ubuntu 18.04 LTS "Bionic Beaver"'],
+              ["ubuntu/focal", 'Ubuntu 20.04 LTS "Focal Fossa"'],
+            ],
+            kernels: {
+              ubuntu: {
+                bionic: [
+                  ["ga-18.04", "bionic (ga-18.04)"],
+                  ["ga-18.04-lowlatency", "bionic (ga-18.04-lowlatency)"],
+                  ["hwe-18.04", "bionic (hwe-18.04)"],
+                  ["hwe-18.04-edge", "bionic (hwe-18.04-edge)"],
+                  ["hwe-18.04-lowlatency", "bionic (hwe-18.04-lowlatency)"],
+                  [
+                    "hwe-18.04-lowlatency-edge",
+                    "bionic (hwe-18.04-lowlatency-edge)",
+                  ],
+                ],
+                focal: [
+                  ["ga-20.04", "focal (ga-20.04)"],
+                  ["ga-20.04-lowlatency", "focal (ga-20.04-lowlatency)"],
+                ],
+              },
+            },
+            default_osystem: "ubuntu",
+            default_release: "focal",
+          },
+          errors: {},
+          loaded: true,
+          loading: false,
+        },
+      },
+      machine: {
+        errors: {},
+        loading: false,
+        loaded: true,
+        items: [
+          {
+            system_id: "abc123",
```

Comment:
Would be good to use the machine test factory here so we can make a start on moving away from partial state objects.

Reply:
Done!